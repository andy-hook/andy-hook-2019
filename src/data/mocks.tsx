import { SocialMeta, Projects } from "@custom-types/model"
import { FluidObject } from "gatsby-image"

export const mockSocialIcons: SocialMeta = {
  email: {
    label: "Email",
    url: "email address",
    icon: "mail",
  },
  twitter: {
    label: "Twitter",
    url: "path/to/social/profile",
    icon: "twitter",
  },
  instagram: {
    label: "Instagram",
    url: "path/to/social/profile",
    icon: "instagram",
  },
  linkedin: {
    label: "Linkedin",
    url: "path/to/social/profile",
    icon: "linkedin",
  },
  dribbble: {
    label: "Dribbble",
    url: "path/to/social/profile",
    icon: "dribbble",
  },
  github: {
    label: "Github",
    url: "path/to/social/profile",
    icon: "github",
  },
}

export const mockProjectsData: Projects = {
  bright: {
    label: "Bright",
    desc: "Bright description",
    images: "bright",
    path: "/projects/bright",
    details: {
      role: "role",
      location: "location",
      date: "date",
    },
  },
  brandwatch: {
    label: "Brandwatch",
    desc: "Brandwatch description",
    images: "brandwatch",
    path: "/projects/brandwatch",
    details: {
      role: "role",
      location: "location",
      date: "date",
    },
  },
  monster: {
    label: "Monster",
    desc: "Monster description",
    images: "monster",
    path: "/projects/monster",
    details: {
      role: "role",
      location: "location",
      date: "date",
    },
  },
  cymbiosis: {
    label: "Cymbiosis",
    desc: "Cymbiosis description",
    images: "cymbiosis",
    path: "/projects/cymbiosis",
    details: {
      role: "role",
      location: "location",
      date: "date",
    },
  },
  jamieson: {
    label: "Jamieson",
    desc: "Jamieson description",
    images: "jamieson",
    path: "/projects/jamieson",
    details: {
      role: "role",
      location: "location",
      date: "date",
    },
  },
  experiments: {
    label: "Experiments",
    desc: "Experiments description",
    images: "experiments",
    path: "/projects/experiments",
    details: {
      role: "role",
      location: "location",
      date: "date",
    },
  },
}

export const mockFluidImageObject: FluidObject = {
  aspectRatio: 1.5,
  sizes: "(max-width: 1000px) 100vw, 1000px",
  src: "/static/4a7de89fcee5d7cd97c940895602fbe0/2f7e7/cover-image.jpg",
  srcSet:
    "/static/4a7de89fcee5d7cd97c940895602fbe0/4d406/cover-image.jpg 250w,↵/static/4a7de89fcee5d7cd97c940895602fbe0/32ee9/cover-image.jpg 500w,↵/static/4a7de89fcee5d7cd97c940895602fbe0/2f7e7/cover-image.jpg",
}
