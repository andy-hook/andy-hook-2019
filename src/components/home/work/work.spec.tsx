import React from "react"
import { mockProjectsData } from "@data/mocks"

import Work from "./work"
import { renderWithTheme } from "@test-utils"

jest.mock("@hooks/page-transition")

jest.mock(
  "@components/shared/projects-grid/projects-grid.container",
  () => "ProjectsGridContainer"
)

describe("<Work />", () => {
  it("renders correctly", () => {
    const tree = renderWithTheme("dark", <Work projects={mockProjectsData} />)
    expect(tree).toMatchSnapshot()
  })
})
