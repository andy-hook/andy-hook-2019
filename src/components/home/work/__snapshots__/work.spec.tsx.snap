// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Work /> renders correctly 1`] = `
<div
  className="sc-ifAKCX kQKtwC"
>
  <div
    className="sc-bdVaJa bUDcPf"
  >
    <div
      className="sc-htpNat dzobEj"
    >
      <div
        className="sc-EHOje dzKsym"
      >
        <h1
          className="sc-bZQynM jBjNHF"
        >
          Projects
        </h1>
        <h2
          className="sc-gzVnrw cpBZxR"
        >
          A selection of my favouriteÂ work
        </h2>
      </div>
      <ProjectsGridContainer />
    </div>
  </div>
</div>
`;
